File: src/main/java/ClassNameProvider.java
import com.intellij.openapi.components.*;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

@State(name = "IntermediaryToNamed", storages = @Storage(StoragePathMacros.CACHE_FILE))
public class ClassNameProvider implements PersistentStateComponent<ClassNameProvider.State> {

    public static class State {
        public Map<String, String> namedToIntermediary = new HashMap<>();

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            State state = (State) o;
            return Objects.equals(namedToIntermediary, state.namedToIntermediary);
        }

        @Override
        public int hashCode() {
            return Objects.hash(namedToIntermediary);
        }
    }

    public State state = new State();

    @Override
    public State getState() {
        return state;
    }

    @Override
    public void loadState(@NotNull State state) {
        this.state = state;
    }

    public static ClassNameProvider getInstance() {
        return ServiceManager.getService(ClassNameProvider.class);
    }

}

File: src/main/kotlin/idea/ClassNameProvider.kt
package cloak.idea

import com.intellij.openapi.components.*
import com.intellij.util.xmlb.XmlSerializerUtil

//// So we can find methods that have intermediary descriptors while the user sees named descriptors
//@State(name = "IntermediaryToNamed", storages = [Storage(StoragePathMacros.CACHE_FILE)])
// class ClassNameProvider : PersistentStateComponent<ClassNameProvider.State> {
//    private var state = State()
//
//    data class State @JvmOverloads constructor(val namedToIntermediary: MutableMap<String, String> = mutableMapOf())
//
//    override fun getState(): State {
//        return state
//    }
//
//    override fun loadState(state: State) {
//        this.state = state
//    }
//
//    companion object {
//        val Instance: ClassNameProvider
//            get() = ServiceManager.getService(ClassNameProvider::class.java)
//    }
//}